corehq/apps/linked_domain/local_accessors.py

```diff
@@ -10,6 +10,40 @@
 from corehq.apps.users.models import SQLUserRole
 from corehq.apps.users.views.mobile import UserFieldsView
 from corehq.apps.integration.models import DialerSettings, GaenOtpServerSettings, HmacCalloutSettings
+from corehq.apps.reports.models import TableauServer, TableauVisualization
+
+
+def get_tableau_server_and_visualizations(domain):
+    server, created = TableauServer.objects.get_or_create(domain=domain)
+    visualizations = TableauVisualization.objects.all().filter(domain=domain).order_by('pk')
+    vis_list = []
+    if not len(visualizations):
+        vis = TableauVisualization(domain=domain, server=server)
+        vis.save()
+        vis_list.append({
+            'domain': domain,
+            'server': server,
+            'view_url': vis.view_url,
+        })
+    else:
+        for vis in visualizations:
+            vis_list.append({
+                'domain': domain,
+                'server': server,
+                'view_url': vis.view_url,
+            })
+    return {
+        'server': {
```

Comment:
Is there only ever one tableau server per domain?

Reply:
That was my understanding. Could be wrong, though.

Reply:
one server per domain, but there could be many visualizations

Reply:
Cool, this looks good then.

@stephherbers It'd be a nice touch to enforce that - you can add `unique=True` to the domain attribute in the model and that'll add a database-level constraint.